internal/api/v1/applications.go

```diff
@@ -73,16 +72,71 @@ func (hc ApplicationsController) Create(w http.ResponseWriter, r *http.Request)
 		return AppAlreadyKnown(createRequest.Name)
 	}
 
+	// Sanity check the services, if any. IOW anything to be bound
+	// has to exist now.  We will check again when the application
+	// is deployed, to guard against bound services being removed
+	// from now till then. While it should not be possible through
+	// epinio itself (*), external editing of the relevant
+	// resources cannot be excluded from consideration.
+	//
+	// (*) `epinio service delete S` on a bound service S will
+	//      either reject the operation, or, when forced, unbind S
+	//      from the app.
+
+	var theIssues []APIError
+
+	for _, serviceName := range createRequest.Configuration.Services {
+		_, err := services.Lookup(ctx, cluster, org, serviceName)
+		if err != nil {
+			if err.Error() == "service not found" {
+				theIssues = append(theIssues, ServiceIsNotKnown(serviceName))
+				continue
+			}
+
+			theIssues = append([]APIError{InternalError(err)}, theIssues...)
+			return MultiError{theIssues}
+		}
+	}
+
+	if len(theIssues) > 0 {
+		return MultiError{theIssues}
+	}
+
+	// Arguments found OK, now we can modify the system state
+
 	err = application.Create(ctx, cluster, appRef, username)
 	if err != nil {
 		return InternalError(err)
 	}
 
-	err = jsonResponse(w, models.ResponseOK)
+	// [INSTANCES CODING]
+	// Save instance information - Force default if not specified
+	// ATTENTION: == 0 -- Use default number of instances
+	//          :  > 0 -- Desired + 1, reduce by one to get actual value.
```

Comment:
- The `nil` value of integers is `0`.
  - I did not really wanted to open the can of worms of allowing even a single negative number and its effect on validation

Using `*integer` instead of `integer` for the field ... Ok, while it felt like a hack I am now unsure why I thought it would be more of a hack than the coding I am using now.

So, we could go and make the `Instances` field a pointer to integer. Then we can use the `nil` as the signal, and a regular int value.

I will work on that now.

Reply:
Done, commit 45ac3302f6d0dd381f5a2004da18101a3df2cb99